### ISSUE TITLE
Move to Syncthing's icons for the tray icon

### ISSUE THREAD

**Issue by dragons4life** [canton7/SyncTrayzor#140]
On windows 10
<img src='http://i.imgur.com/ZL2TDMz.png'/>

Not really a huge bug or problem, but in case you want to fix it.

**Comment by canton7** [comment]
Yeah, that's the result of scaling a large icon down to a small size. Ideally small icons need to be designed to be small, rather than taking an icon which was designed to be large and then scaling it down. 

Unfortunately I don't have the skills to do this. If you or any graphic designers you know 

 it really needs a graphic designer to edit the smaller sizes

**Comment by dragons4life** [comment]
Can you give me the Icon and size you need?

**Comment by canton7** [comment]
Everything's in the repository, here: https://github.com/canton7/SyncTrayzor/tree/master/src/SyncTrayzor/Icons

**Comment by canton7** [comment]
The tray icons are 16x16

**Comment by gerbilOFdoom** [comment]
How's this?
![default_tray ico](https://cloud.githubusercontent.com/assets/1455077/9924596/ec036ea8-5cce-11e5-9ddc-6897580b6aab.png)

**Comment by canton7** [comment]
It's nice and clear, thanks! My worry is that it departs too far from Syncthing's logo: the circle being turned into a square, etc. It's not immediately recognisable as the Syncthing logo, or at least it wasn't to me.

**Comment by canton7** [comment]
Might be worth looking at what the other Gui apps have done in terms of 16x16 icons and asking nicely if they mind us stealing one, heh

**Comment by AudriusButkevicius** [comment]
It looks like the letter D

**Comment by canton7** [comment]
I've little doubt that it's possible to have a 16x16 image which accurately depicts the syncthing logo: the main thing that needs to happen is widening of the white lines, and some tweaking of the anti-aliassing around the edge of the blue circle (although I'm no graphic designer) .

**Comment by gerbilOFdoom** [comment]
Some of my googling revealed that, depending on your build process, the ico file may not be linked correctly.

**Comment by canton7** [comment]
Not sure what you mean by that. The ico file was generated using the gimp, and is not part of the build process.

**Comment by gerbilOFdoom** [comment]
How is the ico file linked to the executable?

**Comment by canton7** [comment]
It's either kept as a separate file, or is an embedded resource, I can't remember.  I'm still not sure what your point is though: the icon looks as it should: the image from which the ico is generated looks exactly like the icon in the taskbar. It's just that the starting image didn't look great...

**Comment by gerbilOFdoom** [comment]
The starting image is actually rather clear - what might be happening is that the wrong layer of the ico file is being used

**Comment by canton7** [comment]
I've looked at this previously (there's an old issue on this) and the correct ico resolution is being used. The issue is that the 16x16 export of the syncthing logo svg, without further tweaking, does not look optimal.

**Comment by canton7** [comment]
Either way, the icon resolution to use is chosen at runtime, not build-time, and had nothing to do with the build process.

**Comment by fredericseiler** [comment]
Best results with the [Lanczos algorithm](https://en.wikipedia.org/wiki/Lanczos_algorithm).
- 96x96 : ![syncthing-96x96](https://cloud.githubusercontent.com/assets/954168/12947910/0cacf982-cfff-11e5-8c1c-11dece7e23a4.png)
- 32x32 : ![syncthing-32x32](https://cloud.githubusercontent.com/assets/954168/12947911/0cb30282-cfff-11e5-9796-5c81a30efc55.png)
- 16x16 : ![syncthing-16x16](https://cloud.githubusercontent.com/assets/954168/12947912/0cb4e886-cfff-11e5-8f35-40d46a48ff6d.png)
- Icon file (remove the .txt on saving) : [syncthing.ico.txt](https://github.com/canton7/SyncTrayzor/files/124759/syncthing.ico.txt)

You also can grab the clean favicon.png from a Syncthing instance (with a bonus vector logo : /assets/img/safari-pinned-tab.svg).

**Comment by canton7** [comment]
Amazing, thankyou! :smile: 

I don't suppose you could generate a 24x24, or tell me which tool you used to generate them?

**Comment by canton7** [comment]
Actually, I'll need to generate a load of others - warning triangle, paused icon, etc.

**Comment by fredericseiler** [comment]
Added some more.

I just used the [RealFaviconGenerator API](http://realfavicongenerator.net/api/non_interactive_api#.VrtSeFjhCUk) to compare compressions and render the .ico.

For the pngs, I grabbed the original 512x512 transparent .png from Wikipedia and resized it using [Paint.NET](http://www.getpaint.net/index.html).

**Comment by fredericseiler** [comment]
There's also a bunch of free online services to generate .ico files :

- [ConvertICO](http://convertico.com/) ([syncthing-convertico.ico.txt](https://github.com/canton7/SyncTrayzor/files/124987/syncthing-convertico.ico.txt) - 361KB)
- [ICO Convert](http://icoconvert.com/) ([syncthing-icoconvert.ico.txt](https://github.com/canton7/SyncTrayzor/files/124985/syncthing-icoconvert.ico.txt) - 41.8KB)

![image](https://cloud.githubusercontent.com/assets/954168/12951625/75490b52-d013-11e5-834e-bc51b8f19791.png)

**Comment by canton7** [comment]
Aha, didn't realise that Paint.NET supported Lanczos, thanks. Looking further, it seems Gimp does too (just not Inkscape, which is what I use to edit the svgs).

I generate the icos with Gimp, so that's fine.

**Comment by fredericseiler** [comment]
The resampling algorithm used by Paint.NET for "Best Quality" preset on resizing is Fant (and it's not so bad).

Anyway, thanks for this really useful app.

**Comment by canton7** [comment]
I guess anything's better than Inkscape's algorithm... I'll get some better icos together using Lanczos. Thanks!

**Comment by canton7** [comment]
Righty, I re-created all of the icons using lanczos, available here: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.1.7.315/artifacts

To be honest it's not much better...

**Comment by fredericseiler** [comment]
Haven't you forgotten to update [default_tray.ico](https://github.com/canton7/SyncTrayzor/blob/develop/src/SyncTrayzor/NotifyIcon/TaskbarIconResources.xaml#L13) in d8ed49a ?

**Comment by canton7** [comment]
Oops! Fixed. Thanks.

**Comment by fredericseiler** [comment]
![image](https://cloud.githubusercontent.com/assets/954168/13139287/2e9e7344-d62c-11e5-8035-7807e9084a02.png)

So you have to choose between blurry and/or pixelated.

Like you [originally said](https://github.com/canton7/SyncTrayzor/issues/140#issuecomment-139464901), it's more a "scale down" issue than a technical/algorithm one. I'll tweak the white ring/paths width to test, because it's fun and I got nothing to do for the next hour.

**Comment by fredericseiler** [comment]
Ok, first attempt :
![image](https://cloud.githubusercontent.com/assets/954168/13142362/c825b09e-d63c-11e5-8153-40ca7c4a3575.png)

**Comment by canton7** [comment]
I'm wary of straying too far from the Syncthing logo (as in, anything that looks visually different to the Syncthing logo): https://forum.syncthing.net/t/syncthing-gtk-icon/6705/4

**Comment by fredericseiler** [comment]
Well, let's hope Syncthing will adopt a simpler logo one day :)

**Comment by scienmind** [comment]
I was wondering about the blurriness on Win10 as well. 
I thought it had something to do with HiDPI scaling. It may be unrelated, but if you are doing something there, it worth taking it into account.

**Comment by canton7** [comment]
Nope, not doing anything there.

**Comment by Noguai** [comment]
I think the problem is more that the dots and lines from the logo do not align with the pixel grid and get too thin when scaled down that much.

How about these?
![small-logos](https://cloud.githubusercontent.com/assets/10562439/15031957/402ad0c2-125d-11e6-9915-cc622079d64f.png)
(Directly exported from Inkscape, no tricks with downscaling etc. involved.)

**Comment by norgeous** [comment]
Will you be supporting / switching to the [new icon set](https://github.com/syncthing/syncthing/tree/master/assets/statusicons)?

**Comment by canton7** [comment]
Maybe. I was going to let it stew in Syncthing for a few weeks, see if anything gets tweaked (since creating all the icos I need from the SVGs is a PITA, and I only want to do it once).

**Comment by canton7** [comment]
I'll probably need to add a few more on top of Syncthing's standard set, but that should be fine.

**Comment by canton7** [comment]
There are a few other issues around changing the icons as well (but to different things). Any attempt should also consider those
